Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MediaUnit.java

```diff
@@ -0,0 +1,207 @@
+    public MediaUnit() {
+    }
+
+    /**
+     * Constructor for developing a media unit from a METS {@code <div>}
+     * element.
+     * 
+     * @param div
+     *            METS {@code <div>} element to be evaluated
+     * @param mets
+     *            the Mets structure is searched for corresponding uses
+     * @param mediaVariants
+     *            list of media variants from which the media variant for the
+     *            given use is taken
+     */
+    MediaUnit(DivType div, Mets mets, Map<String, MediaVariant> mediaVariants) {
+        mediaFiles = div.getFptr().parallelStream().map(Fptr::getFILEID).filter(FileType.class::isInstance)
+                .map(FileType.class::cast)
+                .collect(Collectors.toMap(
+                    file -> mediaVariants.get(mets.getFileSec().getFileGrp().parallelStream()
+                            .filter(fileGrp -> fileGrp.getFile().contains(file)).findFirst()
+                            .orElseThrow(() -> new IllegalArgumentException(
+                                    "Corrupt file: <mets:fptr> not referenced in <mets:fileGrp>"))
+                            .getUSE()),
+                    file -> new MediaFile(file.getFLocat().get(0))));
+        order = div.getORDER().intValue();
+        orderlabel = div.getORDERLABEL();
+    }
+
+    /**
+     * Returns the list of available media variants with the corresponding media
+     * files.
+     * 
+     * @return available media variants with corresponding media files
+     */
+    @Override
+    public Set<Entry<MediaVariant, MediaFile>> getAllUsesWithFLocats() {
+        return mediaFiles.entrySet();
+    }
+
+    /**
+     * Returns the media file for a particular media variant.
+     * 
+     * @param mediaVariant
+     *            media variant for which the media file is to be returned
+     * @return media file for the media variant, can be {@code null} if there is
+     *         no file for the requested variant
+     */
+    @Override
+    public FLocatXmlElementAccessInterface getFLocatForUse(UseXmlAttributeAccessInterface mediaVariant) {
+        return mediaFiles.get(mediaVariant);
+    }
+
+    /**
+     * Returns the order number for this media unit.
+     * 
+     * @return the order number
+     */
+    @Override
+    public int getOrder() {
+        return order;
+    }
+
+    /**
+     * Returns the order label for this media unit.
+     * 
+     * @return the order label
+     */
+    @Override
+    public String getOrderlabel() {
+        return orderlabel;
+    }
+
+    /**
+     * Adds a media file for a specific usage variant.
+     * 
+     * @param mediaVariant
+     *            variant of the added file
+     * @param mediaFile
+     *            media file to add
+     */
+    @Override
+    public void putFLocatForUse(UseXmlAttributeAccessInterface mediaVariant,
+            FLocatXmlElementAccessInterface mediaFile) {
+        mediaFiles.put((MediaVariant) mediaVariant, (MediaFile) mediaFile);
```

Comment:
Please avoid casts.
If this method only accepts MediaVariant and MediaFile please show it in the signature

Reply:
This is not possible because of the interface. The method gets passed classes that implement an interface of the API. Internally I need the classes. This is owed to the API and therefore is not possible otherwise. If you want to change it, these classes would have to be in the API module.

Reply:
Okay, if you say so. How should that look?

Reply:
It would demand a bit of the additional work, so I would keep it for future. I can tell you tomorrow how it could look, so you could consider this change :)